lib/spack/spack/version.py

```diff
@@ -223,6 +248,27 @@ def __str__(self):
     def concrete(self):
         return self
 
+    def _numeric_lt(self, other):
+        """Compares two versions, knowing they're both numeric"""
+        # Standard comparison of two numeric versions
+        for a, b in zip(self.version, other.version):
+            if a == b:
+                continue
+            else:
+                # Numbers are always "newer" than letters.
+                # This is for consistency with RPM.  See patch
+                # #60884 (and details) from bugzilla #50977 in
+                # the RPM project at rpm.org.  Or look at
+                # rpmvercmp.c if you want to see how this is
+                # implemented there.
+                if type(a) != type(b):
+                    return type(b) == int
```

Comment:
If neither `type(a)` nor `type(b)` are `int`, this would give the same result on both `a < b` and `b < a` : is this what we want ?